nl/nl_linux.go

```diff
@@ -338,8 +349,9 @@ func (s *NetlinkSocket) Receive() ([]syscall.NetlinkMessage, error) {
 	if nr < syscall.NLMSG_HDRLEN {
 		return nil, fmt.Errorf("Got short response from netlink")
 	}
-	rb = rb[:nr]
-	return syscall.ParseNetlinkMessage(rb)
+	data := make([]byte, nr)
```

Comment:
Yeah, that's what's bad about `syscall.ParseNetlinkMessage`. It holds references to original slice in returned arguments and original slice won't be collected by garbage collector until those references not disappear. I'm not sure, maybe you can tell, but seems like typical netlink responses are much smaller than `syscall.Getpagesize()` and I show in benchmark that at least for `Add` and `Del` it's true. I can test `nr` and see if it's worth to create new slice and return original to pool, or just use it and never return to pool if you think that it's worth it.
Do you know some netlink call with massive response so I can write benchmark for that case.